libshortfin/python/array_binding.cc

```diff
@@ -254,65 +411,73 @@ void BindArray(py::module_ &m) {
           [](storage &self, bool read, bool write, bool discard) {
             int access = 0;
             if (read) access |= IREE_HAL_MEMORY_ACCESS_READ;
-            if (write) access |= IREE_HAL_MEMORY_ACCESS_WRITE;
+            if (write || discard) access |= IREE_HAL_MEMORY_ACCESS_WRITE;
             if (discard) access |= IREE_HAL_MEMORY_ACCESS_DISCARD;
```

Comment:
Should we have an assert/warning for read=True as well as discard=True? Since contents are undefined, reading seems like undefined behavior.

Reply:
According to Ben, this is actually a valid distinction. There are configurations where knowing that the memory should only be writeable is different from read/write, even with discard. Something about knowing that it is write-only might allow the memory system to configure the caches in a way that penalizes read. So read|write|discard would be used to signal that you intend to read the contents after you've written them and need that to be efficient.